### ISSUE TITLE
__new__ on abstract classes produces false positive "Cannot instantiate abstract class"

### ISSUE THREAD

**Issue by kedder** [python/mypy#14106]
**Bug Report**

When an abstract class defines `__new__` (note, not `__init__`) method, `mypy` produces error when that class is instantiated, even though `__new__()` does not imply instantiation.

**To Reproduce**

```python
from abc import ABCMeta, abstractmethod


class Abstract(metaclass=ABCMeta):
    def __new__(cls) -> "Abstract":
        return super().__new__(Concrete)

    @abstractmethod
    def hello(self) -> int:
        ...


class Concrete(Abstract):
    def hello(self) -> int:
        return 0


x = Abstract()
print(x.hello())
```

**Expected Behavior**

No errors reported when checked with mypy. Calling a class with `__new__` method shuld be effectively treated as just calling a simple method, like 

```python
def make_abstract() -> "Abstract":
    return Concrete()
``` 

**Actual Behavior**

error message is printed:

```
 error: Cannot instantiate abstract class "Abstract" with abstract attribute "hello"  [abstract]
```

**Your Environment**

- Mypy version used: mypy-0.991
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.10.6

**Comment by andy-slac** [comment]
I'm seeing the same issue with Python 3.11 and mypy 1.5.1, this discussion has more details: https://github.com/python/typing/discussions/1487
